# M4 — Profilage des segments

Identifier les caractéristiques et préférences de chaque segment issu de M3 (personas).

In [1]:
import sys
from pathlib import Path
sys.path.append(str(Path.cwd().parent))

import pandas as pd

segments = pd.read_csv("../data/processed/segments.csv")
segments

,Customer_ID,Age,Total_Spent,Nb_Achats,Categorie_Preferee,Cluster,Cluster_Hierarchique,PCA1,PCA2
0,2001,28,500.0,2,Clothing,2,2,-0.539820,1.669480
1,2002,35,750.0,1,Clothing,0,0,1.934829,-0.281477
2,2003,22,300.0,1,Outerwear,1,1,-1.938821,-0.886414
3,2004,30,600.0,1,Accessories,0,0,0.543812,-0.501588


## Statistiques par segment

In [2]:
segments.groupby("Cluster")[["Age", "Total_Spent", "Nb_Achats"]].agg(["mean", "count"])

Age       Total_Spent       Nb_Achats      
         mean count        mean count      mean count
Cluster                                              
0        32.5     2       675.0     2       1.0     2
1        22.0     1       300.0     1       1.0     1
2        28.0     1       500.0     1       2.0     1

In [3]:
segments.groupby("Cluster")["Categorie_Preferee"].value_counts()

Cluster  Categorie_Preferee
0        Clothing              1
         Accessories           1
1        Outerwear             1
2        Clothing              1
Name: count, dtype: int64

## Personas

Générés automatiquement à partir des statistiques par cluster ci-dessus (à affiner manuellement avec un volume de données réel).

In [4]:
for cluster_id, group in segments.groupby("Cluster"):
    print(f"--- Segment {cluster_id} ({len(group)} client(s)) ---")
    print(f"Âge moyen : {group['Age'].mean():.0f} ans")
    print(f"Dépense moyenne : {group['Total_Spent'].mean():.0f} $")
    print(f"Fréquence d'achat moyenne : {group['Nb_Achats'].mean():.1f}")
    print(f"Catégorie préférée : {group['Categorie_Preferee'].mode().iloc[0]}")
    print()

--- Segment 0 (2 client(s)) ---
Âge moyen : 32 ans
Dépense moyenne : 675 $
Fréquence d'achat moyenne : 1.0


Catégorie préférée : Accessories

--- Segment 1 (1 client(s)) ---
Âge moyen : 22 ans
Dépense moyenne : 300 $
Fréquence d'achat moyenne : 1.0
Catégorie préférée : Outerwear

--- Segment 2 (1 client(s)) ---
Âge moyen : 28 ans
Dépense moyenne : 500 $
Fréquence d'achat moyenne : 2.0
Catégorie préférée : Clothing



### Synthèse des personas (à interpréter avec prudence — n=4 clients segmentés sur les 5 de l'échantillon, un client n'ayant aucun achat enregistré)

| Segment | Profil | Comportement | Recommandation marketing |
|---|---|---|---|
| 0 | Clients ~30 ans, dépense élevée (~675$), achats peu fréquents (1x) | Achat ponctuel à forte valeur unitaire | Cibler avec des offres premium / cross-sell plutôt que des relances fréquence |
| 1 | Client plus jeune (~22 ans), dépense plus faible (~300$) | Entrant / petit budget | Programme de fidélisation progressif pour augmenter la valeur dans le temps |
| 2 | Client multi-achats (2x), dépense moyenne (~500$) | Engagé, achats répétés | Prioritaire pour la rétention et les recommandations personnalisées |
